creator-node/test/snapbackSM.test.js

```diff
@@ -810,4 +811,274 @@ describe('test SnapbackSM', function () {
+        },
+        'http://cnWithSpId3.co': {
+          successRate: 0.1,
+          successCount: 1,
+          failureCount: 9
+        }
+      }
+    }
+    const { requiredUpdateReplicaSetOps, potentialSyncRequests } = await snapback.aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers, userSecondarySyncMetricsMap)
+
+    // Make sure that the CN with low sync success put into `requiredUpdateReplicaSetOps`
+    assert.strictEqual(requiredUpdateReplicaSetOps.length, 1)
+    assert.strictEqual(requiredUpdateReplicaSetOps[0].unhealthyReplicas.size, 1)
+    assert.ok(requiredUpdateReplicaSetOps[0].unhealthyReplicas.has('http://cnWithSpId3.co'))
+    assert.strictEqual(potentialSyncRequests.length, 1)
+    assert.strictEqual(potentialSyncRequests[0].endpoint, 'http://cnWithSpId2.co')
+  })
+
+  it('[selectRandomReplicaSetNodes] replace 1 unhealthy replica', async function () {
+    const { SnapbackSM: mockSnapback } = proxyquire(
+      '../src/snapbackSM/snapbackSM.js',
+      {
+        './StateMachineConstants': {
+          MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS: 10_000
+        }
+      }
+    )
+    const snapback = new mockSnapback(nodeConfig, getLibsMock())
+
+    const healthyReplicaSet = new Set([
+      'http://healthyCn1.co',
+      'http://healthyCn2.co'
+    ])
+    const numberOfUnhealthyReplicas = 1
+    const healthyNodes = [
+      'http://healthyCn1.co',
+      'http://healthyCn2.co',
+      'http://healthyCn4.co',
+    ]
+    const wallet = '0x00fc5bff87afb1f15a02e82c3f671cf5c9ad9e6d'
+
+    // Mock the user having no state on any of the healthy nodes
+    snapback._retrieveClockValueForUserFromReplica = async (_, __) => -1
+
+    const newReplicaSetNodes = await snapback.selectRandomReplicaSetNodes(
+      healthyReplicaSet,
+      numberOfUnhealthyReplicas,
+      healthyNodes,
+      wallet
+    )
+
+    // The missing node in healthyReplicaSet should've been replaced with a node from healthyNodes
+    assert.strictEqual(newReplicaSetNodes.length, 1)
+    assert.ok(healthyNodes.includes(newReplicaSetNodes[0]))
+    assert.ok(!healthyReplicaSet.has(newReplicaSetNodes[0]))
+  })
+
+  it('[selectRandomReplicaSetNodes] replace 2 unhealthy replicas and ignore healthy node with state', async function () {
+    const { SnapbackSM: mockSnapback } = proxyquire(
+      '../src/snapbackSM/snapbackSM.js',
+      {
+        './StateMachineConstants': {
+          MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS: 10_000
+        }
+      }
+    )
+    const snapback = new mockSnapback(nodeConfig, getLibsMock())
+
+    const healthyReplicaSet = new Set([
+      'http://healthyCn1.co'
+    ])
+    const numberOfUnhealthyReplicas = 2
+    const healthyNodes = [
+      'http://healthyCn1.co',
+      'http://healthyNodeWithUserState.co',
+      'http://healthyCn4.co',
+      'http://healthyCn5.co',
+    ]
+    const wallet = '0x00fc5bff87afb1f15a02e82c3f671cf5c9ad9e6d'
+
+    // Mock the user having no state on any of the healthy nodes
```

Comment:
this comment is not accurate for this test or the one below

Reply:
fixed